cmd/talosctl/cmd/talos/edit.go

```diff
@@ -0,0 +1,168 @@
+	"github.com/talos-systems/talos/pkg/machinery/api/machine"
+	"github.com/talos-systems/talos/pkg/machinery/client"
+)
+
+var editCmdFlags struct {
+	namespace string
+}
+
+// editCmd represents the edit command.
+var editCmd = &cobra.Command{
+	Use:     "edit <RESOURCE/NAME>",
+	Aliases: []string{"apply"},
+	Short:   "Edit a resource from the default editor.",
+	Args:    cobra.RangeArgs(1, 2),
+	Long: `The edit command allows you to directly edit any API resource 
+you can retrieve via the command line tools. 
+
+It will open the editor defined by your TALOS_EDITOR, 
+or EDITOR environment variables, or fall back to 'vi' for Linux 
+or 'notepad' for Windows.`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return WithClient(func(ctx context.Context, c *client.Client) error {
+			edit := editor.NewDefaultEditor([]string{
+				"TALOS_EDITOR",
+				"EDITOR",
+			})
+
+			resourceType := args[0]
+			if resourceType != "config" {
+				return fmt.Errorf("only the config resource can be edited")
+			}
+
+			var lastError string
+
+			editFn := func(parentCtx context.Context, msg client.ResourceResponse) error {
+				if msg.Resource == nil {
+					return nil
+				}
+
+				if msg.Metadata.GetError() != "" {
+					fmt.Fprintf(os.Stderr, "%s: %s\n", msg.Metadata.GetHostname(), msg.Metadata.GetError())
+
+					return fmt.Errorf("aborted")
+				}
+
+				for {
+					var (
+						buf bytes.Buffer
+						w   io.Writer = &buf
+						id  string
+					)
+
+					metadata := msg.Resource.Metadata()
+
+					if metadata != nil {
+						id = metadata.ID()
+					}
+
+					if runtime.GOOS == "windows" {
+						w = crlf.NewCRLFWriter(w)
+					}
+
+					_, err := w.Write([]byte(
+						fmt.Sprintf(
+							"# Editing %s/%s at host %s\n", resourceType, id, msg.Metadata.GetHostname(),
+						),
+					))
+					if err != nil {
+						return err
+					}
+
+					if lastError != "" {
+						lines := strings.Split(lastError, "\n")
+
+						_, err = w.Write([]byte(
+							fmt.Sprintf("#\n# %s\n", strings.Join(lines, "\n# ")),
+						))
+						if err != nil {
+							return err
+						}
+					}
+
+					body, err := yaml.Marshal(msg.Resource.Spec())
+					if err != nil {
+						return err
+					}
+
+					_, err = w.Write(body)
+					if err != nil {
+						return err
+					}
+
+					edited, _, err := edit.LaunchTempFile(fmt.Sprintf("%s-%s-edit-", resourceType, id), ".yaml", &buf)
+					if err != nil {
+						return err
+					}
+
+					editedWithoutComments := bytes.TrimSpace(cmdutil.StripComments(edited))
+
+					if len(bytes.TrimSpace(editedWithoutComments)) == 0 {
+						fmt.Println("Apply was skipped: empty file.")
+
+						break
+					}
+
+					if bytes.Equal(
+						editedWithoutComments,
+						bytes.TrimSpace(cmdutil.StripComments(body)),
+					) {
+						fmt.Println("Apply was skipped: no changes detected.")
+
+						break
+					}
+
+					_, err = c.ApplyConfiguration(parentCtx, &machine.ApplyConfigurationRequest{
+						Data:      edited,
+						Immediate: true,
```

Comment:
it might be amazing to support both `immediate: false` and `true` updates

if we could add a flag `--imediate` here it will be just a killer feature ;)

Reply:
sure

Reply:
I've added it and made it true by default. Let me know if that doesn't work

Reply:
can we make it `false` by default to align with `apply-config` for now?